### PR TITLE
[v2] Modifying profiles to have correct comma placements within comments

### PR DESCRIPTION
**Pull request by jace-roell**
**What It Does**
Applies a comma to the end of all values in profile kvp except the last value
Applies a comma to all commented profile types except base

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/core/src/apiml/Services.ts
Comment (traeok) [inline packages/core/src/apiml/Services.ts:229]: Can this comment on line 229 be removed? Looks like its no longer being used.

File: packages/core/src/apiml/Services.ts
Comment (traeok) [inline packages/core/src/apiml/Services.ts:215]: `endComma` can be made `const` with the use of the ternary operator:

```suggestion
            // add comma to end of kvp depending on if base profile
            const endComma = key.includes("base") ? "" : ",";
```

File: packages/core/src/apiml/Services.ts
Comment (traeok) [inline packages/core/src/apiml/Services.ts:226]: Not required, but `kvPair` can be assigned to the result of `elements.reduce`, which would allow `kvPair` to be reassigned as a constant variable:

```suggestion
            const kvPair = elements.reduce((all, current: string, index) => {
                return all.concat(index === elements.length - 1 ? `\n//"${key}": "${element}"` :  `\n//"${key}": "${element}"${endComma}`);
            }, "");
```

**Review by traeok — COMMENTED** [review]
Looks good, thanks @jace-roell! I left a couple comments regarding an unused line and variable declarations.

File: packages/core/src/apiml/Services.ts
Comment (jace-roell) [reply packages/core/src/apiml/Services.ts]: comment removed

**Review by awharn — COMMENTED** [review]
The changes look good to me, but it appears the branch is out of date. Will approve once `master` is merged into this branch and any conflicts are resolved.

**Review by traeok — APPROVED** [review]
LGTM, thanks @jace-roell for the enhancement and for quickly addressing feedback! 😁

**Review by awharn — CHANGES_REQUESTED** [review]
Looks good to me, but there appears to be a lint warning. Once that is fixed, I will approve. 😄

**Review by traeok — APPROVED** [review]

**Review by awharn — APPROVED** [review]

File: packages/core/CHANGELOG.md
Comment (anaxceron) [inline packages/core/CHANGELOG.md:7]: - Can we add a modifier to Services.ts to explain what it is?: "Modified the property/profile/biscuit/whatever Services.ts to ..."
- Please add a period after "commas."

**Review by anaxceron — APPROVED** [review]
Left a comment about the Changelog item, nothing major.

**Review by gejohnston — APPROVED** [review]

**Pull request merged**
